primitives/transition/context.go

```diff
@@ -28,71 +28,116 @@ import (
 
 // Context is the context for the state transition.
 type Context struct {
-	context.Context
+	// consensusCtx is the context passed by CometBFT callbacks
+	// We pass it down to be able to cancel processing (although
+	// currently CometBFT context is set to TODO)
+	consensusCtx context.Context
+	// consensusTime returns the timestamp of current consensus request.
+	// It is used to build next payload and to validate currentpayload.
+	consensusTime math.U64
+	// Address of current block proposer
+	proposerAddress []byte
 
-	MeterGas bool
-	// OptimisticEngine indicates whether to optimistically assume
-	// the execution client has the correct state certain errors
-	// are returned by the execution engine.
-	OptimisticEngine bool
-	// SkipPayloadVerification indicates whether to skip calling NewPayload
-	// on the execution client. This can be done when the node is not
+	// verifyPayload indicates whether to call NewPayload on the
+	// execution client. This can be done when the node is not
 	// syncing, and the payload is already known to the execution client.
-	SkipPayloadVerification bool
-	// SkipValidateRandao indicates whether to skip validating the Randao mix.
-	SkipValidateRandao bool
-	// SkipValidateResult indicates whether to validate the result of
+	verifyPayload bool
+	// verifyRandao indicates whether to validate the Randao mix.
+	verifyRandao bool
+	// verifyResult indicates whether to validate the result of
 	// the state transition.
-	SkipValidateResult bool
-	// Address of current block proposer
-	ProposerAddress []byte
-	// ConsensusTime returns the timestamp of current consensus request.
-	// It is used to build next payload and to validate currentpayload.
-	ConsensusTime math.U64
+	verifyResult bool
+
+	// meterGas controls whether gas data related to the execution
+	// layer payload should be meter or not. We currently meter only
+	// finalized blocks.
+	meterGas bool
+	// optimisticEngine indicates whether to optimistically assume
+	// the execution client has the correct state certain errors
+	// are returned by the execution engine.
+	optimisticEngine bool
+}
+
+func NewTransitionCtx(
+	consensusCtx context.Context,
+	time math.U64,
+	address []byte,
+) *Context {
+	return &Context{
+		consensusCtx:    consensusCtx,
+		consensusTime:   time,
+		proposerAddress: address,
+
+		// by default we don't meter gas
+		// (we care only about finalized blocks gas)
+		meterGas: false,
+
+		// by default we don't have optimistic engine
+		// as it basically mute some checks
+		optimisticEngine: false,
+
+		// by default we keep all verification
+		verifyPayload: true,
+		verifyRandao:  true,
+		verifyResult:  true,
```

Comment:
I rather set all validators to true by default. I think the safest is to explicitly switch them off